src/main/java/com/ibm/watson/nlu/DemoServlet.java

```diff
@@ -142,12 +153,26 @@ private boolean getConfigParams() {
 			return false;
 		}
 
-		// if one isn't set, consider them all not set
-		if (username.equals("<add_nlu_username>") ||
-				password.equals("<add_nlu_password>") ||
-				modelId.equals("<add_model_id>")) {
+		// make sure we have what we need
+		if (modelId.equals("<add_model_id>")) {
 			return false;
 		}
+
+		if (baseURL.equals("<add_nlu_url>")) {
+			return false;
+		}
+
+		if ((apikey.equals("add_nlu_iam_apikey")) &&
+			(username.equals("<add_nlu_username>") || password.equals("<add_nlu_password>"))) {
+			return false;
+		}
+
+		if (apikey.equals("add_nlu_iam_apikey")) {
+			useIamApiKey = false;
+		}
+
+		logger.info("using IAM_APIKEY = " + useIamApiKey);
```

Comment:
the message should be updated, at first i thought it would print an API key, but it's printing a boolean. there is no variable called `IAM_APIKEY`. it's `useIamApiKey` or `apikey`. The caps make me think you're referencing an environment variable, too.